Ouster Lidar

sacrover opened the pull request:
Notes:
1. Refactored code to run with Ouster OS064 Lidar
2. The Lidar now launches from trailbot_bringup.launch.py instead of launching from the slam_3D.launch.py.  This ensures all hardware lifecycle nodes are initialized properly before SLAM, CV files are launched. 
3. The Cartographer and Nav2 tuning params are based on 1 indoor and 1 outdoor test. 
4. The bagging_files scripts are **NOT** yet updated with Ouster  Lidar in this PR. (TBD: once Logitech camera PR is merged)
5. `dev` branch is created off of `marc/integration `

nickoenig37 committed:
OS0-64 Lidar

added ros-drivers
cartographer unstable

nickoenig37 committed:
2D Cartographer Fix

sacrover committed:
ouster-carto3D

all nodes sat
carto params needs tuning

sacrover committed:
ouster-stable

IMU, LiDAR, carto, nav2 - SAT
carto, nav2 - not tuned
LiDAR urdf link corrected

sacrover committed:
ouster imu setup

changed topics/frames to use ouster imu data
removed external imu_link
tuned cartogrpher (based on indoor-feb16 bag)

sacrover committed:
nav2 params update

-ouster launch is renamed to trailbot bringup (includes driving and other hardware initialization)

MarcAntoineLavoie commented on nav/config/nav2_params_points.yaml:
```diff
@@ -139,131 +139,101 @@ controller_server:
+          topic: /ouster/scan
+          obstacle_max_range: 35.0
+          obstacle_min_range: 0.0
+          raytrace_max_range: 2.5
```
why is the raytrace max range substantially smaller now?

MarcAntoineLavoie commented on nav/config/nav2_params_points.yaml:
```diff
@@ -139,131 +139,101 @@ controller_server:
   #   ros__parameters:
   #     use_sim_time: True
 
-map_server:
```
Why are we dropping all of these parameters? Are they just default values?

MarcAntoineLavoie commented on nav/config/nav2_params_points.yaml:
```diff
@@ -139,131 +139,101 @@ controller_server:
+local_costmap:
+  local_costmap:
     ros__parameters:
       update_frequency: 1.0
```
Why are we updating / publishing the costmap at 1Hz? Isn't this a bit slow?

MarcAntoineLavoie commented on human_detection/human_detection/human_detection_node.py:
```diff
@@ -223,7 +223,7 @@ def __init__(self,parser_args,model,configs):
         self.lidar_subscription = self.create_subscription(
             PointCloud2,
-            'velodyne_points',  
+            './._points',  
```
is this the actual name of the topic?

MarcAntoineLavoie commented on slam/config/trailbot_lds_3d.lua:
```diff
@@ -32,24 +32,29 @@ options = {
-POSE_GRAPH.constraint_builder.sampling_ratio = 0.03 --originally 0.3
+-- POSE_GRAPH.optimization_problem.huber_scale = 5e2 --originally 100 
+POSE_GRAPH.optimization_problem.huber_scale = 100
+POSE_GRAPH.optimize_every_n_nodes = 50 --originally 90
```
Is this required? Shouldn't we want more frequent optimization like in the original config?

MarcAntoineLavoie reviewed (changes requested):
Mostly questions about the choice of parameters in the configs

sacrover committed:
param updates

-changed max_raytrace_range
-changed local costmap update frequency
-corrected lidar subscription topic in human detection node
